node/subsystem/src/messages.rs

```diff
@@ -526,12 +526,12 @@ pub enum ProvisionerMessage {
 }
 
 impl ProvisionerMessage {
-	/// If the current variant contains the relay parent hash, return it.
-	pub fn relay_parent(&self) -> Option<Hash> {
+	/// Returns the relay parent this message is assigned to.
+	pub fn relay_parent(&self) -> Hash {
```

Comment:
Except where you have changed it in this PR, all `*Message::relay_parent()` returns `Option<Hash>`. It would be nice to maintain consistency across the interface, even if we don't abstract it into a trait.

Reply:
We could also remove the other ones if they are not required.

I see this as any message to a subsystem that spawns jobs needs to provide a relay parent. While messages to long running subsystems don't need to provide a relay parent.

Reply:
This means, we can either remove the other functions or just keep it the way it is done currently.

Reply:
Right, I see what you mean. I don't have any strong objections to just removing `fn relay_parent` from those messages which don't need it.